src/main/java/com/nike/cerberus/service/AuthenticationService.java

```diff
@@ -272,6 +284,15 @@ private IamRoleAuthResponse authenticate(IamPrincipalCredentials credentials, Ma
         return iamRoleAuthResponse;
     }
 
+    private AuthTokenResponse stsAuthenticate(String iamPrincipalArn, Map<String, String> authPrincipalMetadata) {
```

Comment:
Does this need to be a separate method? Why not put this code in the other `stsAuthenticata`